ModulemdTranslationHelpers/Utils.py

```diff
@@ -58,3 +63,94 @@ def get_translation_catalog_from_index(index, project_name):
         catalog.add(translatable_string, locations=locations)
 
     return catalog
+
+
+def get_modulemd_translations_from_catalog(catalogs, index):
+    # Nested dictionary `data` contains information from catalog like:
+    # Key_1: <module>:<stream>
+    # Value_1: Key_2: <locale>
+    #          Value_2: list of (<string_type>:<msg.string>)
+
+    data = defaultdict(defaultdict)
+
+    for catalog in catalogs:
+        for msg in catalog:
+            for location, _ in msg.locations:
+                (module_name, stream_name, string_type,
+                 profile_name) = split_location(location)
+
+                key_1 = ("{}:{}").format(module_name, stream_name)
+                if profile_name is None:
+                    value_2 = ("{}:{}").format(string_type, msg.string)
+                else:
+                    value_2 = ("{}:{}:{}").format(
+                        string_type, profile_name, msg.string)
+
+                # Adding into data
+                data[key_1].setdefault(
+                    catalog.locale_identifier, []).append(value_2)
+
+    for mod_stream, translations in data.items():
+        (module_name, stream_name) = mod_stream.split(":")
+        now = datetime.utcnow()
```

Comment:
Don't do this multiple times through the loop. Asking for the current time is expensive. Get the time exactly once at the start of processing and use the same value everywhere.